playbooks/fundamentals/Container_Image_Build_Guidelines.adoc

```diff
@@ -0,0 +1,139 @@
+The *1st RUN* line executes package installs from one line separated by semicolons due to how a docker image is layered.  We will explain that in detail later in our guide and why that is important to consider.
+The *2nd RUN* line is adding some content to our web server.
+The *EXPOSE* line allows traffic to our web server on port 80 of the container.
+The *ENTRYPOINT* line is used to signify the command that will be run as if we were on the command line and this is the main execution process.
+The *CMD* line should be used to run the software contained by our image, along with any arguments. 
+The *USER* line should be used to run the httpd process as a non-root user.
+
+== Onto the advanced use case and best practices
+
+=== Size does matter
+
+In our above example we started with a very basic RHEL 7 image, which should be obtained from a trusted source. Images vary in size and what we started with (at the time of writing this document) is large in comparison to an Atomic Host image found https://access.redhat.com/containers/?tab=overview#/registry.access.redhat.com/rhel7-atomic[here] ( ~ 29M using 87 RPMs ) vs the rhel7 base image found https://access.redhat.com/containers/?tab=overview#/registry.access.redhat.com/rhel[here] ( ~ 71M using 152 RPMs ).  One attribute to consider is the size of the image.  Our design and intent should be taken into account to see which style of image best fits our use case.  An additional consideration is Atomic images do not include SUID programs which give temporary permissions to a user to run a program/file with the permissions of the file owner rather that the user who runs it which is an additional consideration that falls under security. The package installer, which is dnf and not yum, is a package manager that does not have python as a requirement and thus is lighter weight and has a reduction in the rpm’s in the image itself which carries extra benefits as a reduced attack serfice as well.  
+
+=== Base image
+
+Once we have landed on our image style it’s time to consider additional items such as where we actually obtain the image from.  It is paramount that we trust the source that provides the base image to us.  Images from Red Hat are a trusted source and maintained with constant improvement and patches.  
+
+=== Security
+
+Our third consideration to our image standard is security.  Security has different areas of concern when implementing container images.  We mentioned the trusted source as one such aspect as well as patching for vulnerabilities, but we also have to consider who is ultimately going to run our container.  When installing packages to the base image we will need to be root to accomplish such tasks, but once those tasks that require root privilege are complete, we should change the running container user with a USER 1001 or similar line in our file before we run the container.  
```

Comment:
the ability to build images that operate properly without privileges is maybe the most important skill we can transfer. some more discussion of that topic and some examples would be very impactful